src/main/java/edu/ucsb/mapache/documents/Channel.java

```diff
@@ -0,0 +1,181 @@
+import java.util.Objects;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Document(collection = "channels")
+public class Channel {
+    private String id;
+    private String name;
+    private String creator;
+    private Boolean is_archived;
+    private Boolean is_general;
+    private List<String> members;
+    private ChannelTopic topic;
+    private ChannelPurpose purpose;
+
+    public Channel() {
+    }
+
+    public Channel(String id, String name, String creator, Boolean is_archived, Boolean is_general, List<String> members, ChannelTopic topic, ChannelPurpose purpose) {
+        this.id = id;
+        this.name = name;
+        this.creator = creator;
+        this.is_archived = is_archived;
+        this.is_general = is_general;
+        this.members = members;
+        this.topic = topic;
+        this.purpose = purpose;
+    }
+
+    public String getId() {
+        return this.id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getCreator() {
+        return this.creator;
+    }
+
+    public void setCreator(String creator) {
+        this.creator = creator;
+    }
+
+    public Boolean isIs_archived() {
+        return this.is_archived;
+    }
+
+    public Boolean getIs_archived() {
+        return this.is_archived;
+    }
+
+    public void setIs_archived(Boolean is_archived) {
+        this.is_archived = is_archived;
+    }
+
+    public Boolean isIs_general() {
+        return this.is_general;
+    }
+
+    public Boolean getIs_general() {
+        return this.is_general;
+    }
+
+    public void setIs_general(Boolean is_general) {
+        this.is_general = is_general;
+    }
+
+    public List<String> getMembers() {
+        return this.members;
+    }
+
+    public void setMembers(List<String> members) {
+        this.members = members;
+    }
+
+    public ChannelTopic getTopic() {
+        return this.topic;
+    }
+
+    public void setTopic(ChannelTopic topic) {
+        this.topic = topic;
+    }
+
+    public ChannelPurpose getPurpose() {
+        return this.purpose;
+    }
+
+    public void setPurpose(ChannelPurpose purpose) {
+        this.purpose = purpose;
+    }
+
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this)
+            return true;
+        if (!(o instanceof Channel)) {
+            return false;
+        }
+        Channel channel = (Channel) o;
+        return Objects.equals(id, channel.id) && Objects.equals(name, channel.name) && Objects.equals(creator, channel.creator) && Objects.equals(is_archived, channel.is_archived) && Objects.equals(is_general, channel.is_general) && Objects.equals(members, channel.members) && Objects.equals(topic, channel.topic) && Objects.equals(purpose, channel.purpose);
```

Comment:
We should use `EqualsBuilder` over `Object.equals` because it results in less branches to test.

Reply:
Done